### ISSUE TITLE
Use GitHub as official source of Prelude?

### ISSUE THREAD

**Issue by Gabriel439** [dhall-lang/dhall-lang#162]
IPFS has proven to be a slow and unreliable host for the Dhall Prelude.  I have to restart and redeploy the IPFS mirror for the Dhall Prelude every 1-2 weeks and even when the mirror works the download speed is poor.

Now that we have an official GitHub repository for the Dhall Prelude, I'm wondering if we should just use GitHub as the official source of the Prelude?  In other words:

```bash
$ dhall <<< 'https://raw.githubusercontent.com/dhall-lang/Prelude/e44284bc37a5808861dacd4c8bd13d18411cb961/Natural/sum'
```
```haskell
∀(xs : List Natural) → Natural

  λ(xs : List Natural)
→ List/fold Natural xs Natural (λ(l : Natural) → λ(r : Natural) → l + r) 0
```

My main reasons for proposing the transition are:

* It's much more reliable
* It's faster
* It's free

... and it seems to have basic feature parity with IPFS:

* You can pin to specific revision (using the SHA) or to the latest revision (using `master`)
* Users can create a local clone using `git clone`

**Comment by ocharles** [comment]
You get even shorter URLs if you cut releases of the prelude and used tags, too.

**Comment by Profpatsch** [comment]
Or just use a cloned git repo. :)

**Comment by f-f** [comment]
I agree with the proposal 💯 

However, Github (unlike IPFS) is centralized, and goes down "often" (IMHO), so we should have a B-plan as well.
IPFS would be ok, and/or we could [mirror the repo to Gitlab](https://docs.gitlab.com/ee/workflow/repository_mirroring.html).
The additional benefit of the mirror on GItlab over IPFS is that it would still be on git, so you would need to only change the remote in case of outage.

**Comment by arianvp** [comment]
I'm very curious about what the exact reliability issues were. Did pinning fail? As in, IPFS would lose the fact that you pinned the files?

**Comment by f-f** [comment]
> How about a local filesystem cache to paper over outages? This would permit development where internet is unavailable or untrustworthy (airplanes, coffee shops) after enough things are cached

Yeah, that works for local development, but when I wrote that I was thinking more of CI builds. In that case having the build pickup an alternative url or path location would be the best option (as #163 details).
In our current deployment we have the Prelude as a submodule of the infrastructure repo (so the tradeoff is that CI fetches it for every build, but we have nice relative paths), and in case of outage we would either re-host it on our internal git server, or just bake it into the Docker image we use for CI.

**Comment by Gabriel439** [comment]
@blast-hardcheese: See https://github.com/dhall-lang/dhall-lang/issues/129 for a proposal for local filesystem caching

@arianvp: The pin works correctly, but IPFS has a memory leak which requires restarting and redeploying the system.  If the machine is unresponsive the pin is uneffective

**Comment by f-f** [comment]
Fixed in #177
